World.py

```diff
@@ -159,6 +161,7 @@ def __init__(self, id, settings):
                     raise Exception('Custom hint text too large for %s', loc['location'])
                 self.hint_text_overrides.update({loc['location']: loc['text']})
 
+        self.item_hints+=self.item_added_hint_types["named-item"]
```

Comment:
Recommend changing this to `self.item_hints = settings.item_hints + self.item_added_hint_types["named-item"]` to prevent the item duplication issue in the spoiler. This will also make the key in the spoiler match the plando file if used.

Reply:
I was going to test this and suggest it this morning. Thanks!

Reply:
Ah, this does wipe out the spoiler `item_hints` though. Is that okay?

Reply:
Since the `item_hints` list is pre-defined by the hint distribution anyway, it's not a big deal to not have access to it in the spoiler log. This also prevents any potential issues of feeding back the spoiler log to the generator as a plando file, so it works out.

The only nice thing having the item_hints in the spoiler log would do is tell you for certain which of the items in the list the generator picked to hint, though that isn't really a big deal to not always know .

Reply:
Excellent then.